### ISSUE TITLE
Nested route params not working

### ISSUE THREAD

**Issue by EugenEistrach** [TanStack/router#2498]
### Which project does this relate to?

Router

### Describe the bug

When creating a route with nested dynamic parents we always render the parent route.

### Your Example Website or App

https://stackblitz.com/edit/tanstack-router-oxxjed?file=app%2Froutes%2Fposts.%24postId.tsx

### Steps to Reproduce the Bug or Issue

1. Go to any post detail
2. There is an link to navigate to nested url with dynamic param (/posts/$postId/$someOtherId)

### Expected behavior

We should render the nested route not the parent. 

### Platform

- OS: macos
- Browser: chrome

**Comment by EugenEistrach** [comment]
I found a workaround to get it working. E.g instead of using the same path name for the parent path if I use something different I will be able to navigate to the route. E.g instead of calling the file:
`posts.$postId.$someOtherId.tsx`
I can call it:
`posts.$postIdOther.$someOtherId.tsx`

and then I'm able to navigate it. If this is not a bug its atleast unintuitive behaviour as in other frameworks we can just nest such routes.

**Comment by EugenEistrach** [comment]
Ok I'm sorry, I misunderstood some of the behaviour and all routes are able to be nested it seems. If we put  <Outlet /> into the parent route it will render accordingly.

**Comment by jackmillward** [comment]
I've just been burnt by this and now have multiple files that refer to the same ID but I've had to use different names, so this will become a bit of a pain in the future to maintain and keep things consistent.

An example of my project's routing:
```
/quizzes
/quizzes/$quizId
/quizzes/$quizId/play
```

This, naturally, resulted in a file layout that looks like:
```
src/routes/_pages/quizzes/index.tsx
src/routes/_pages/quizzes/$quizId/index.tsx
src/routes/_pages/quizzes/$quizId/play.tsx
```

I also tried with:
```
src/routes/_pages/quizzes/index.tsx
src/routes/_pages/quizzes/$quizId.tsx
src/routes/_pages/quizzes/$quizId.play.tsx
```

Either way results in the following:
![Image](https://github.com/user-attachments/assets/038315de-df17-43e0-b224-532bb6258707)

I've put logs in the loaders in `$quizId.tsx` and in `$quizId.play.tsx`. The router, currently, seems to:
```
1. Hit `$quizId.play.tsx` and runs through the loader.
2. Goes up to the parent route.
3. Does not run the loader there.
4. Renders the component from the parent route (`$quizId.tsx`).
5. Ignores the component in `$quizId.play.tsx`.
```

The current fix as the above user mentioned is to change that parameter name to something different, then it works fine.

Is this expected behaviour? Because I agree with @EugenEistrach , it's not particularly intuitive and will make codebases quite difficult to maintain in future.
